### PR TITLE
Build Debian Package on Xenial and Stretch

### PR DESCRIPTION
**Pull request by dwightguth**
This should expand the OSes we support to include Ubuntu 16.04, Ubuntu 18.04, and Debian Stretch. Next PR I will investigate Arch Linux and CentOS 7. But I wanted to get the refactorings to the pipeline reviewed first.

### CODE REVIEW

**Review by traiansf — COMMENTED** [review]
Although changes look good, I don't feel qualified to review this.  I guess I could try to test that k works on the distributions we mention if you think that would help.

**Comment by dwightguth** [comment]
The PR actually tests K on those distributions. I'm fine with Everett reviewing though.

**Review by traiansf — APPROVED** [review]

**Review by ehildenb — CHANGES_REQUESTED** [review]
Things look good except that it's clogging up the CI server severely. Lets reduce the parallelism to keep that from happening.

File: Dockerfile [hunk 1]
```diff
@@ -13,8 +17,23 @@ RUN groupadd -g $GROUP_ID user && \
...
+
+ADD --chown=user:user llvm-backend/src/main/native/llvm-backend/matching/stack.yaml /home/user/.tmp-haskell2/
+ADD --chown=user:user llvm-backend/src/main/native/llvm-backend/matching/package.yaml /home/user/.tmp-haskell2/
+ADD --chown=user:user llvm-backend/src/main/native/llvm-backend/matching/submodules/kore/stack.yaml /home/user/.tmp-haskell2/submodules/kore/
+ADD --chown=user:user llvm-backend/src/main/native/llvm-backend/matching/submodules/kore/src/main/haskell/kore/package.yaml /home/user/.tmp-haskell2/submodules/kore/src/main/haskell/kore/
+RUN    cd /home/user/.tmp-haskell2 \
+    && stack build --only-snapshot --test
+
+ADD pom.xml ktree/pom.xml llvm-backend/pom.xml haskell-backend/pom.xml ocaml-backend/pom.xml kernel/pom.xml java-backend/pom.xml k-distribution/pom.xml kore/pom.xml /home/user/.tmp-maven/
+RUN mvn dependency:go-offline
```
Comment (ehildenb) [inline Dockerfile]: Should this be `cd ...`ing to the correct directory first?

**Review by ehildenb — COMMENTED** [review]
Looks good, except for one thing I'm wondering about.

**Review by ehildenb — APPROVED** [review]

**Pull request merged**